internal/cmd/dockerbuild/main.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/bufbuild/buf/private/bufpkg/bufplugin/bufpluginref"
+	"github.com/bufbuild/buf/private/pkg/interrupt"
+	"github.com/bufbuild/plugins/internal/plugin"
+	"golang.org/x/sync/errgroup"
+)
+
+// dockerbuild is a helper program used to build plugins from Dockerfiles in an optimized fashion.
+// It replaces some clunky (and non-parallel) jobs in the Makefile.
+// It knows about relationships between common containers (like protoc/grpc Bazel plugins),
+// which enables optimized builds which build common code first before binaries.
+// It also makes it easier to add new labels to images using existing code to parse buf.plugin.yaml.
+
+// larger amount of parallelism lead to OOM errors in testing - clamp for now
+const maxLimit = 8
```

Comment:
If I let this run with 10 cores (m1) I'd see random times where the process would get killed trying to build all images. Dialed this down smaller to hopefully avoid any memory OOM killer.